app/models/user.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  attr_accessor :remember_token
+  validates :name, presence: true, uniqueness: { case_sensitive: false }
+  has_many :events
+  has_many :attendees
+  scope :event_ids, ->(id) { Attendee.joins(:attended_events).select(:event_id).where('user_id == ?', id) }
+  scope :upcoming, ->(id) { Event.where('id IN (?) and date >= ?', User.event_ids(id), Date.today) }
```

Comment:
This could be written better in the Event model as `scope :upcoming, -> { where('date >= ?', Time.zone.now) }` same goes for past